azure-iot-hub-devicesdk/samples/receive_device_method.py

```diff
@@ -0,0 +1,83 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import os
+from six.moves import input
+import logging
+import threading
+from azure.iot.hub.devicesdk import DeviceClient
+from azure.iot.hub.devicesdk.auth.authentication_provider_factory import from_connection_string
+
+logging.basicConfig(level=logging.ERROR)
+
+
+# The connection string for a device should never be stored in code. For the sake of simplicity we're using an environment variable here.
+conn_str = os.getenv("IOTHUB_DEVICE_CONNECTION_STRING")
+# The "Authentication Provider" is the object in charge of creating authentication "tokens" for the device client.
+auth_provider = from_connection_string(conn_str)
+# For now, the SDK only supports MQTT as a protocol. the client object is used to interact with your Azure IoT hub.
+# It needs an Authentication Provider to secure the communication with the hub, using either tokens or x509 certificates
+device_client = DeviceClient.from_authentication_provider(auth_provider, "mqtt")
+
+
+# connect the client.
+device_client.connect()
+
+
+# define behavior for handling methods
+def method1_listener(device_client):
+    while True:
+        method_call = device_client.receive_method("method1")  # Wait for method1 calls
+        result = True  # set some result
```

Comment:
>True [](start = 17, length = 4)

I understand this might be hard to enforce in Python but I think to be valid a method response payload has to be valid json (to be confirmed). the samples should convey this

Reply:
We definitely can't enforce it, but if the expected payload is a json I can make the samples contain that.

---
In reply to: [266186000](https://github.com/Azure/azure-iot-sdk-python-preview/pull/61#discussion_r266186000) [](ancestors = 266186000)